jax/interpreters/pxla.py

```diff
@@ -613,35 +613,19 @@ def parallel_callable(fun, backend, axis_name, axis_size, global_axis_size,
     raise ValueError("'devices' argument to pmap must be non-empty, or None.")
 
   # Determine global_axis_size for use in AxisEnv.
+  must_run_on_all_devices = True
   if devices:
     assert global_axis_size is None  # Checked in api.py
     global_axis_size = len(devices)
-  elif xb.host_count() > 1:
-    if global_axis_size is None:
-      # TODO(skye): relax this constraint or provide functionality for
-      # automatically passing appropriate `devices`.
-      # TODO(trevorcai): This check forces us to provide global_axis_size for
-      # all pmaps on pmap-on-pod. Can we do it after tracing?
-      if axis_size != xb.local_device_count():
-        raise ValueError(
-            "On multi-host platforms, the input to pmapped functions must have "
-            "leading axis size equal to the number of local devices if no "
-            "`devices` argument is specified. Got axis_size=%d, "
-            "num_local_devices=%d" % (axis_size, xb.local_device_count()))
-      global_axis_size = xb.device_count()
+    must_run_on_all_devices = False
   else:
-    if global_axis_size is not None:
-      if global_axis_size != axis_size:
-        raise ValueError(
-            "Specified axis_size {} doesn't match received axis_size {}.".format(
-                global_axis_size, axis_size))
-    else:
-      global_axis_size = axis_size
-
-  log_priority = logging.WARNING if FLAGS.jax_log_compiles else logging.DEBUG
```

Comment:
Probably don't want to lose the logging from jax_log_compiles, consider relocating to another spot.

Reply:
Think you left it out of the commit, not seeing the new logging.

Reply:
Oof. I'm copy/pasting internal/external code back and forth, and keep screwing up this logging! (I actually tried to move it from the start but kept failing somehow.) I think it's actually there now...